bentoml/_internal/bento/build_config.py

```diff
@@ -373,19 +536,49 @@ def _python_options_structure_hook(d: t.Any, _: t.Type[PythonOptions]):
 bentoml_cattr.register_structure_hook(PythonOptions, _python_options_structure_hook)
 
 
-def _dict_arg_converter(  # type: ignore
-    options_type: t.Type[t.Union[DockerOptions, CondaOptions, PythonOptions]]
-):
+Options: t.TypeAlias = t.Union[DockerOptions, CondaOptions, PythonOptions]
+
+
+class OptionsFactory:
```

Comment:
I think it makes more sense to do `Options.with_defaults` instead of having this class.